moped-editor/src/views/projects/projectView/ProjectComponents/utils.js

```diff
@@ -0,0 +1,66 @@
+import { useMemo } from "react";
+import booleanIntersects from "@turf/boolean-intersects";
+import circle from "@turf/circle";
+
+/* Filters a feature collection down to one type of geometry */
+export const useFeatureTypes = (featureCollection, geomType) =>
+  useMemo(() => {
+    const features = featureCollection.features.filter((feature) => {
+      const thisGeom = feature.geometry.type.toLowerCase();
+      return thisGeom.includes(geomType);
+    });
+    return { type: "FeatureCollection", features };
+  }, [featureCollection, geomType]);
+
+/*
+Bit of a hack to generate intersection labels from nearby streets. 
+this relies on nearby lines being avaialble in-memory, which 
+is not guaranteed. a reliable solution would be query the AGOL streets
+layer on-the-fly to grab street names - not sure if this is worth it TBH 
+*/
```

Comment:
Hmm, I'm torn about doing a GIS operation here in the map, but it is straightforward and certainly going to be valid as it's coming out of `turf`. I wonder what projection its using to allow it to use meters, probably web mercator (3857)?

Reply:
Same here - I didn't look at this code that closely so I think @johnclary could respond better. The only constructive input that I have is that I also trust the output of `turf` 😅

Reply:
yes, thanks—i see three options (this current hack not being one of them):
1. remove this feature entirely
2. add the intersection labels to the underlying dataset w/ help from the GEO team
3. query an AGOL feature service on-the-fly to get the labels.
will make an issue 🤪

Reply:
https://github.com/cityofaustin/atd-data-tech/issues/10632